apps/api/src/app/environments-v1/usecases/construct-framework-workflow/construct-framework-workflow.usecase.ts

```diff
@@ -199,7 +209,25 @@ export class ConstructFrameworkWorkflow {
 
     return foundWorkflow;
   }
+
+  private processSkipOption(controlValues: { [x: string]: unknown }, variables: FullPayloadForRender) {
+    const skipRules = controlValues.skip as RulesLogic<AdditionalOperation>;
+
+    // TODO: check if we need to add more validation here (is empty object, etc.)
+    if (!skipRules) {
+      return false;
+    }
+
+    const { result, error } = evaluateRules(skipRules, variables);
+
+    if (error) {
+      this.logger.error({ err: error }, 'Failed to evaluate skip rule', LOG_CONTEXT);
```

Comment:
Shall we pause workflow execution when there is a skip condition error? I think we should.

Reply:
Good point. It’s better to avoid sending a notification altogether than to send an invalid or unintended one.